source/features/exclude-filter-shortcut.js

```diff
@@ -0,0 +1,51 @@
+/*
+This feature ...
+*/
+
+import {h} from 'dom-chef';
+import select from 'select-dom';
+import delegate from 'delegate';
+
+const qualifier = [
+	'author',
+	'assignee',
+	'label',
+	'review',
+	'reviewed-by'
+];
+
+// TODO: We guess the last qualifier is the right one
+// This might not be always correct, specifically other features could influence this
+// Also if e.g. two labels are selected the alt clicking would remove it and negate the other one
+// To solve this we would need to parse the item to see what filter it would build
+const lastQualifier = new RegExp(`((?:${qualifier.join('|')}):[^:]+)$`);
```

Comment:
This isn't always the case, I had underestimated the complexity of this feature.

Your way with `getLabelName` was better where, for example:

1. The label `bug` is selected
2. The user alt+clicks the `bug` label

What happens there? The regular click would simply drop `label:bug` from the URL so `label:bug` can't be found in the URL.

If `label:bug` isn't there, **add** `-label:bug`
If `label:bug` is there, **replace** it with `-label:bug`
If `-label:bug` is there, **drop** it (to match the regular click behavior)

Reply:
Thanks for the comment, I will get back to this feature soon.

I think I already have tinkered around with detecting which item was clicked but there are a lot of special cases, like the label being surrounded with `"` when it contains a space.

I still think it would be great if this was on Githubs Roadmap, but it is fun to hack around 😄